public/extensions/iframe.js

```diff
@@ -0,0 +1,158 @@
+import daily from "./core.js";
+import flexpanel from "./flexpanel.js";
+import callstate from "./callstate.js";
+
+let fp, iframeEl, locationEl, call;
+let open = false;
+
+/* Defaults */
+let defaultButtons = {
+  selectUrl: {
+    iconPath: "https://www.svgrepo.com/show/107601/globe.svg",
+    label: "Iframe",
+    tooltip: "Open Iframe",
+  },
+};
+
+let props = {
+  url: "about:blank", // used to set the initial URL for the iframe.
+  buttons: defaultButtons, // used to change the button names and/or icons.
+  showUrl: false, // controls whether to display the active iframe URL as a pseudo-address-bar
+  // above the iframe content.
+};
+
+/* Public interface */
+let self;
+export default self = {
+  configure: function (p) {
+    Object.assign(props, p);
+  },
+  open: function (url) {
+    // sets a new URL for the iframe and opens
+    props.url = url;
+    open = true;
+    callstate.updateCallState("iframe", { url, open });
+  },
+  close: function () {
+    // closes the iframe and clears URL
+    props.url = "about:blank";
+    open = false;
+    callstate.updateCallState("iframe", { url: props.url, open });
+  },
+  show: function () {
+    // just shows the iframe (flexpanel)
+    open = true;
+    callstate.updateCallState("iframe", { open });
+  },
+  hide: function () {
+    // just hides without removing URL
+    open = false;
+    callstate.updateCallState("iframe", { open });
+  },
+  setUrl: function (url) {
+    // updates URL without changing
+    // show/hide state
+    props.url = url;
+    callstate.updateCallState("iframe", { url: props.url });
+  },
+};
+
+/* Daily configuration */
+daily.beforeCreateFrame((parentEl, properties) => {
+  // TODO: maybe namespace shared resources like tray buttons?
+  if (!properties.customTrayButtons) {
+    properties.customTrayButtons = {};
+  }
+  Object.assign(properties.customTrayButtons, props.buttons);
+  // create a flexpanel from the passed parentEl, which becomes the daily panel
+  let containerEl = document.createElement("div");
+  containerEl.style = `
```

Comment:
I know it would be cleaner to extract it to a separate stylesheet, but for now I think it's more valuable to have this all self-contained. We'll see how things play out though!